src/main/java/frc/robot/sensors/IrSensor.java

```diff
@@ -0,0 +1,66 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+/**
+ * Code by Charlie Macdonald.
+ * Last modified on Feb 5th, 2020.
+ */
+
+package frc.robot.sensors;
+
+import edu.wpi.first.wpilibj.AnalogInput;
+import edu.wpi.first.wpilibj.MedianFilter;
+import edu.wpi.first.wpilibj.Sendable;
+import edu.wpi.first.wpilibj.smartdashboard.SendableBuilder;
+import edu.wpi.first.wpilibj.smartdashboard.SendableRegistry;
+
+public class IrSensor implements Sendable {
+    private static final int kIRPort = 1; //set IR sensor port
+        //TODO: The IR sensor's port may change, depending on the robot setup. (Use a robot-specific config to set this?)
+
+    //median filter to discard outliers. Filters over 10 samples
+    private final MedianFilter m_filter = new MedianFilter(10);
+
+    private final AnalogInput m_irsensor;
+
+    //Set the formula values to calculate IR sensor distance:
+        //Values for short range (5-30cm) sensor: 12.8528, -0.007545, -3.92, 35, 4.5.
+        //TODO: If a different IR sensor is used, these values will need to be updated. (Use a robot-specific config to set them?)
+    private final double irSlopeValue = 12.8528;
+    private final double irXValue = -0.007545;
+    private final double irConstant = -3.92;
+    private final double irMaxDistance = 35;
+    private final double irMinDistance = 4.5;
```

Comment:
Are any of these tune-able? And if so, would it be beneficial to be able to edit them over NetworkTables without re-deploying the code? If so you may want to look at changing these into FluidConstants.

Reply:
No, these are meant to curve-fit based on the physical response of the sensor. They shouldn't change.

One suggestion: maybe include the units? e.g. irMaxDistanceCm

Reply:
Charlie can you write somewhere in the comments the exact model number of the sensor this class is meant to represent? Sharp model ABC123-whatever. Do you still have the data sheet on your computer? Otherwise I'll help you find that number.

Reply:
Variables won't change often (only if we swap out IR sensors), so there wouldn't be much benefit from changing those into FluidConstants.

Added sensor model number, and added units to variable names.